File: 415. Add Strings/Python/Solution.py
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        i = len(num1) - 1
        j = len(num2) - 1
        result = ''
        carry = 0
        while i >= 0 or j >= 0:
            if i >= 0:
                carry += ord(num1[i]) - ord('0')
            if j >= 0:
                carry += ord(num2[j]) - ord('0')
            result += chr(carry % 10 + ord('0'))
            carry //= 10
            i -= 1
            j -= 1
        if carry == 1:
            result += '1'
        return result[::-1]

File: 415. Add Strings/C++/Solution.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int i = num1.size() - 1;
        int j = num2.size() - 1;
        string result = "";
        int carry = 0;
        while (i >= 0 || j >= 0) {
            if (i >= 0) {
                carry += num1[i] - '0';
            }
            if (j >= 0) {
                carry += num2[j] - '0';
            }
            result += (char)(carry % 10 + '0');
            carry /= 10;
            i--;
            j--;
        }
        if (carry == 1) {
            result += '1';
        }
        reverse(result.rbegin(), result.rend());
        return result;
    }
};